client/src/components/CardDialog/CardDialog.tsx

```diff
@@ -110,71 +108,11 @@ const CardDialog = ({ name, columnId, tag }: DialogProps): JSX.Element => {
-                        content: 'description',
-                        icon: 'contacts',
-                        id: `item-${Math.floor(Math.random() * 999999)}`,
-                      });
-                    }}
-                  >
-                    Description
-                  </Button>
-                  <Button
-                    className={hasItem('deadline') ? classes.columnButtonActive : classes.columnButton}
-                    onClick={() => {
-                      addItem({
-                        title: 'Deadline:',
-                        content: 'deadline',
-                        icon: 'schedule',
-                        id: `item-${Math.floor(Math.random() * 999999)}`,
-                      });
-                    }}
-                  >
-                    Deadline
-                  </Button>
-                  <Button
-                    className={hasItem('comment') ? classes.columnButtonActive : classes.columnButton}
-                    onClick={() => {
-                      addItem({
-                        title: 'Comment:',
-                        content: 'comment',
-                        icon: 'bubble',
-                        id: `item-${Math.floor(Math.random() * 999999)}`,
-                      });
-                    }}
-                  >
-                    Comment
-                  </Button>
-                  <Button
-                    className={hasItem('attachment') ? classes.columnButtonActive : classes.columnButton}
-                    onClick={() => {
-                      addItem({
-                        title: 'Attachments:',
-                        content: 'attachment',
-                        icon: 'attachment',
-                        id: `item-${Math.floor(Math.random() * 999999)}`,
-                      });
-                    }}
-                  >
-                    Attachment
-                  </Button>
-                  <Button
-                    className={hasItem('checklist') ? classes.columnButtonActive : classes.columnButton}
-                    onClick={() => {
-                      addItem({
-                        title: 'Checklist:',
-                        content: 'checklist',
-                        icon: 'checklist',
-                        id: `item-${Math.floor(Math.random() * 999999)}`,
-                      });
-                    }}
-                  >
-                    Check-list
-                  </Button>
+                  <DialogActionButton title="Description" content="description" icon="contacts" />
+                  <DialogActionButton title="Deadline" content="deadline" icon="schedule" />
+                  <DialogActionButton title="Comment" content="comment" icon="bubble" />
+                  <DialogActionButton title="Attachment" content="attachment" icon="attachment" />
+                  <DialogActionButton title="Checklist" content="checklist" icon="checklist" />
```

Comment:
Great job on the refactor here! The above code was bulky plus not `DRY`; this looks significantly better!